src/utils/api.ts

```diff
@@ -0,0 +1,39 @@
+import { parseUserInfo } from '~/parser';
+
+enum HttpMethod {
+  Get = 'Get',
+  Post = 'Post',
+}
+
+interface Option {
+  method: HttpMethod;
+}
+
+const GITHUB_API_BASE_URL = 'https://api.github.com';
+
+const apiPaths = {
+  repositories: (name: string) => `/users/${name}/repos`,
+};
+
+const request = <T>(path: string, option: Option) => {
+  return fetch(`${GITHUB_API_BASE_URL}${path}`, option).then((res) => res.json() as Promise<T>);
+};
+
```

Comment:
분리 하신 걸를 보니 다른 프로젝트에서도 재사용하실 것 같은데
`apiPaths` 처럼 `baseURL`을 분리해도 좋을 것 같아유 ㅎㅎ


```suggestion
const request = (baseURL: string) => <T>(path: string, option: Option) => {
  return fetch(`${baseURL}${path}`, option).then((res) => res.json() as Promise<T>);
};

```

Reply:
오 좋아여!